tests/js/pay2ContractHash.scrypttest.js

```diff
@@ -0,0 +1,164 @@
+
+
+const { expect } = require('chai');
+const { compileContract, newTx, createInputFromPrevTx} = require('../../helper');
+const {  buildContractClass, Bytes, Sig, SigHashPreimage,  bsv, toHex, getPreimage, buildTypeClasses, SigHash, PubKeyHash, PubKey } = require('scryptlib');
+const {  toHashedMap, findKeyIndex, hash160, signTx, buildOpreturnScript } = require('scryptlib/dist/utils');
+
+const inputIndex = 0;
+const inputSatoshis = 100000;
+
+const outputAmount = inputSatoshis
+
+const privateKey = new bsv.PrivateKey.fromRandom('testnet')
+const publicKey = privateKey.publicKey
+const pkh = bsv.crypto.Hash.sha256ripemd160(publicKey.toBuffer())
+
+
+const Counter = buildContractClass(compileContract('advancedCounter.scrypt'));
+
+const Pay2ContractHash = buildContractClass(compileContract('pay2ContractHash.scrypt'));
+
+
+let map = new Map();
+
+const tokenId = 111;
+map.set(tokenId, new PubKeyHash(toHex(pkh)))
+
+// tx that transfering token from PubKeyHash to Counter ContractHash
+const tx1 = newTx(inputSatoshis);
+
+describe('Pay2ContractHash', () => {
+    let counter, pay2ContractHash, counterContractHash
+    before(() => {
+        counter = new Counter(0);
+
+        pay2ContractHash = new Pay2ContractHash(toHashedMap(map));
+
+        counterContractHash = new PubKeyHash(hash160(counter.lockingScript.toHex()));
+      });
+    
+
+    it('should success when token transfer from PubKeyHash to Counter ContractHash', () => {
```

Comment:
`should succeed when token is transferred from a PubKeyHash to Counter's contract hash`

Same elsewhere.